zhmcclient/_partition_link.py

```diff
@@ -0,0 +1,752 @@
+
+    @logged_api_call
+    def list(self, full_properties=False, filter_args=None):
+        """
+        List the partition links known to the HMC.
+
+        Partition links for which the authenticated user does not have
+        object-access permission are not included.
+
+        Any resource property may be specified in a filter argument. For
+        details about filter arguments, see :ref:`Filtering`.
+
+        The listing of resources is handled in an optimized way:
+
+        * If this manager is enabled for :ref:`auto-updating`, a locally
+          maintained resource list is used (which is automatically updated via
+          inventory notifications from the HMC) and the provided filter
+          arguments are applied.
+
+        * Otherwise, if the filter arguments specify the resource name as a
+          single filter argument with a straight match string (i.e. without
+          regular expressions), an optimized lookup is performed based on a
+          locally maintained name-URI cache.
+
+        * Otherwise, the HMC List operation is performed with the subset of the
+          provided filter arguments that can be handled on the HMC side and the
+          remaining filter arguments are applied on the client side on the list
+          result.
+
+        This method performs the "List Partition Links" HMC operation.
+
+        HMC/SE version requirements:
+
+        * SE version >= 2.16.0
+        * for technology-specific support, see the API features described in
+          :ref:`Partition Links`
+
+        Authorization requirements:
+
+        * Object-access permission to any partition links to be included in the
+          result.
+
+        Parameters:
+
+          full_properties (bool):
+            Controls that the full set of resource properties for each returned
+            partition link is being retrieved, vs. only the following short
+            set: "object-uri", "cpc-uri", "name", "state", and "type".
+
+          filter_args (dict):
+            Filter arguments that narrow the list of returned resources to
+            those that match the specified filter arguments. For details, see
+            :ref:`Filtering`.
+
+            `None` causes no filtering to happen.
+
+        Returns:
+
+          : A list of :class:`~zhmcclient.PartitionLink` objects.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        result_prop = 'partition-links'
+        list_uri = self._base_uri
+        return self._list_with_operation(
+            list_uri, result_prop, full_properties, filter_args, None)
+
+    @logged_api_call
+    def create(self, properties=None):
+        """
+        Create a partition link from the input properties.
+
+        The input properties may specify an initial attachment of the partition
+        link to one or more partitions.
+
```

Comment:
For CTC it is required to specify a `paths` array on initial creation describing the physical connectivity between the Ficon adapters (which adapter is connected to which). We may want to state that here (although of course you can find that information also in the HMC WS-API book.

Reply:
Added.